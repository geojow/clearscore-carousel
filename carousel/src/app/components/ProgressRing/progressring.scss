.progress-ring {
  position: absolute;
  height: 370px;
  width: 370px;
}

circle {
  transform: rotate(-90deg);
  transform-origin: 50% 50%;
  animation: 1.4s ease-in-out both circle-animation;
  stroke-linecap: round;
}

@keyframes circle-animation {
  0% {
    stroke-dashoffset: 1112.1237993707869;
  }
}
