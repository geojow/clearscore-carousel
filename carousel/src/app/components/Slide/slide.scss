.slide__container {
  height: 375px;
  width: 375px;
  position: absolute;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  border-radius: 50%;
  font-family: Helvetica, sans-serif;

  color: white
}

p {
  margin: 0;
}

.slide__content-upper, .slide__content-lower {
  font-size: 20px;
  font-weight: 300;
}

.slide__content-middle {
  font-size: 90px;
  font-weight: 100;
}

