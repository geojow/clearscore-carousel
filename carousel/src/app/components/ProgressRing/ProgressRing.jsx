import React from 'react';
import PropTypes from 'prop-types';
import bemHelper from '../../utils/bem';
import './progressring.scss'

const cn = bemHelper({ block: "progress-ring" })

const ProgressRing = ({ progress }) => {

  const radius = 185
  const stroke = 4
  const normalisedRadius = radius - stroke * 2;
  const circumference = normalisedRadius * 2 * Math.PI;

  const strokeDashoffset = circumference - progress / 100 * circumference;

  return (
    <svg className={cn('')}>
      <circle
        fill="transparent"
        strokeWidth={stroke}
        strokeDasharray={circumference}
        style={{ strokeDashoffset, '--test': "black" }}
        r={normalisedRadius}
        cx={radius}
        cy={radius}
      />
    </svg>)
}

ProgressRing.propTypes = {
  progress: PropTypes.number.isRequired
}

export default ProgressRing;
