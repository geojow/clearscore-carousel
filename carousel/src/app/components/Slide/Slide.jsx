import React from 'react'
import bemHelper from '../../utils/bem';
import './slide.scss';

const cn = bemHelper({ block: 'slide' });

const Slide = () => <li className={cn('container')} />

export default Slide;
