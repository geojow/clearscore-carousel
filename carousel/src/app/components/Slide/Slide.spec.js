import React from 'react';
import { shallow } from 'enzyme';
import Slide from './Slide';
import ProgressRing from '../ProgressRing/ProgressRing'

describe('Slide', () => {
  let wrapper;

  beforeEach(() => wrapper = shallow(<Slide slide={{ id: '1', upper: '', middle: '', lower: '', progress: 10 }} />));

  it('should render a <li />', () => {
    expect(wrapper.find('li').length).toEqual(1);
  })

  it('should render a ProgressRing component', () => {
    expect(wrapper.containsMatchingElement(<ProgressRing progress={10} />)).toEqual(true);
  })

  it('should render the content from slide prop', () => {
    wrapper.setProps({ slide: { id: '1', upper: 'Your credit score is', middle: '350', lower: 'out of 700', progress: null } });
    expect(wrapper.text()).toEqual('Your credit score is350out of 700')
  })
})
