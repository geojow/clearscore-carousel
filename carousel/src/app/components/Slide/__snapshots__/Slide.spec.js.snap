// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Slide
    slide={
      Object {
        "id": "1",
        "lower": "",
        "middle": "",
        "progress": 10,
        "upper": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <li
          className="slide__container"
        >
          <p
            className="slide__content-upper"
          >
            
          </p>
          <p
            className="slide__content-middle slide__content-middle--1"
          >
            
          </p>
          <p
            className="slide__content-lower"
          >
            
          </p>
        </li>,
        <ProgressRing
          progress={10}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="slide__content-upper"
            >
              
            </p>,
            <p
              className="slide__content-middle slide__content-middle--1"
            >
              
            </p>,
            <p
              className="slide__content-lower"
            >
              
            </p>,
          ],
          "className": "slide__container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "slide__content-upper",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "slide__content-middle slide__content-middle--1",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "",
              "className": "slide__content-lower",
            },
            "ref": null,
            "rendered": "",
            "type": "p",
          },
        ],
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "progress": 10,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <li
            className="slide__container"
          >
            <p
              className="slide__content-upper"
            >
              
            </p>
            <p
              className="slide__content-middle slide__content-middle--1"
            >
              
            </p>
            <p
              className="slide__content-lower"
            >
              
            </p>
          </li>,
          <ProgressRing
            progress={10}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="slide__content-upper"
              >
                
              </p>,
              <p
                className="slide__content-middle slide__content-middle--1"
              >
                
              </p>,
              <p
                className="slide__content-lower"
              >
                
              </p>,
            ],
            "className": "slide__container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "slide__content-upper",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "slide__content-middle slide__content-middle--1",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
                "className": "slide__content-lower",
              },
              "ref": null,
              "rendered": "",
              "type": "p",
            },
          ],
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "progress": 10,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
